File: pc-agent-plugin/src/core/security/CredentialProvider.ts
import { CredentialError } from "./CredentialErrors";

export interface CredentialProvider {
  readSecret(secretRef: string): Promise<string>;
  upsertSecret(secretRef: string, secret: string): Promise<void>;
  deleteSecret(secretRef: string): Promise<void>;
}

export interface CredentialRecord {
  secretRef: string;
  encryptedSecret: string;
}

export type CredentialFailure = CredentialError;

File: pc-agent/src/main/java/com/chatcui/agent/auth/CredentialProvider.java
package com.chatcui.agent.auth;

public interface CredentialProvider {
    String readSecret(String secretRef);

    void upsertSecret(String secretRef, String secret);

    final class CredentialException extends RuntimeException {
        public enum Reason {
            NOT_FOUND,
            ACCESS_DENIED,
            CORRUPT_ENTRY,
            INVALID_INPUT
        }

        private final Reason reason;

        public CredentialException(Reason reason, String message) {
            super(message);
            this.reason = reason;
        }

        public Reason reason() {
            return reason;
        }
    }
}
